src/utils/jax_utils.py

```
@@ -0,0 +1,212 @@
+        batch_states_jax = jnp.array(batch_states, dtype=jnp.int32)
+
+        # Transfer to device if specified
+        if device is not None:
+            batch_states_jax = device_put(batch_states_jax, device)
+
+        # Initialize for iteration tracking
+        curr_states = batch_states_jax
+        iterations = 0
+
+        # Track state history for cycle detection
+        state_history = {}
```

Ruff (0.11.9)

Local variable `state_history` is assigned to but never used

Remove assignment to unused variable `state_history`

(F841)